### DISCUSSION TITLE
possibility to define the ai_deck wifi as client to connect remote server ?

### DISCUSSION THREAD

**Discussion by mvtkurd** [bitcraze/discussions#94] [Q&A] [answered]
Hi all,

I'm going to send some images from remote server to ai deck. I would like to know if it's possible to define the ai deck wifi as client and connect to the gateway and then server send or receive data through it ? or do you have some examples which I can modify it based on my goal ? like the attached image
![wifi](https://user-images.githubusercontent.com/32674939/173380849-d526fb8f-438e-490b-ae19-15c584c71f48.png)

**Comment by evoggy** [comment, marked as answer]
Hi!

Looking at the description I'm not sure if you want the AI deck to be server or client, but either is possible but currently we only support the AI deck being server. So just like the python script that receives the images, you can write your own app that connects and receives the images.

If you want to do it the other way around (AI deck connects to your server, which you will  probably need since you have the gateway in between) or you want to more advanced commands you will have to implement this yourself. There's lots of examples in the ESP IDF on how to do different protocols (websocket, http etc) and also how to make the ESP connect to a server. Have a look [here](https://github.com/espressif/esp-idf/tree/v4.3.1/examples/protocols)) as a start and then merge it with our code.

**Reply by mvtkurd** [reply]
@evoggy 

Thanks for replay . well let me tell you more about my target maybe you could suggest me better way. I'm going to test reinforcement learning model on ai deck and I have a running simulation of photorealistic environment on the server which provides 10 frames per second. Now I'm going to send that images to the input of neural network running on ai deck and send back the result from ai deck to server. 

there's another option to use my laptop instead of server so that would be great if I can connect ai deck directly to my loptop (UART , SPI , ...) and send/receive data but I'm not professional in that field (C code and firmware) and any example would be appreciated.

**Reply by evoggy** [reply]
Ok, thanks for the clarification. Looking at the description this will require a lot of changes to the system, connecting like the graph above is only part of the problem you need to solve. You also have to make the protocol to send the images to the AI deck to process them. CPX (via WiFi) can be used for the communication, but you will need to add a protocol on top of this to send the images to the GAP8. The link above will give some examples on how to connect to the computer and for the GAP8 related things I would suggest looking at our examples and/or the GreenWaves examples.
